bdlist.go

```diff
@@ -0,0 +1,191 @@
+// Package bdlist implements simple bi-directional list
+// and its typical operations
+package bdlist
+
+// An Element contains value and pointers to the other Elements,
+// if any
+type Element struct {
+	Prev, Next *Element
+	Value      any
+}
+
+// A BDList contains a collection of elements, linked together
+// in bi-directional way
+type BDList struct {
+	Head   *Element // head of the list
+	Tail   *Element // tail of the list
+	Length int      // length of the list
+}
+
+// NewElement creates new Element with given value
+func NewElement(value any) *Element {
+	element := &Element{Value: value, Prev: nil, Next: nil}
+	return element
+}
+
+// New creates new, empty BDList
+func New() *BDList {
+	list := &BDList{Head: nil, Tail: nil, Length: 0}
+	return list
+}
+
+// Append adds an element after Tail
+func (list *BDList) Append(element *Element) {
+	if element != nil {
+		if list.Length == 0 {
+			list.Head = element
+			list.Head.Prev = nil
+			list.Tail = element
+			list.Head.Next = list.Tail
+			list.Tail.Prev = list.Head
+			list.Tail.Next = nil
+			list.Length++
+		} else {
+			list.Tail.Next = element
+			element.Prev = list.Tail
+			element.Next = nil
+			list.Tail = list.Tail.Next
+			list.Length++
+		}
+	}
```

Comment:
I think we can shorten this code :)

```
if element == nil {
   return err // we should throw error here or sth
}
if list.Head == nil and list.Tail == nil {
   list.Head = element
   list.Tail = element
} else {
  list.Tail.Next = element
  element.Prev = list.Tail
 list.Tail = list.Tail.next
}
list.Length++
```
I'm not sure if checking/resettings pts here to nil is necessary